frontend/test/specs/comment/commentControllerSpec.js

```diff
@@ -0,0 +1,584 @@
+'use strict';
+
+(describe('Test CommentController', function() {
+    beforeEach(module('app'));
+    
+    const POSTS_URI = "/api/posts";
+    let commentCtrl, scope, httpBackend, mdDialog;
+    let commentService, messageService;
+    
+    const user = {
+        name: 'name',
+        key: 'user-key',
+        state: 'active',
+        current_institution: {name: 'inst', key:'inst-key'}
+    };
+
+    let nilReply = {};
+    let reply = {
+        text: "reply", 
+        id: 1,
+        likes: []
+    };
+
+    let comment = {
+        text: "comment",
+        id: 5,
+        data_comments: [],
+        replies: {},
+        author_key: user.key,
+        likes: []
+    };
+
+    const post = {
+        title: 'post principal', author_key: user.key,
+        key: "123456", comments: "/api/posts/123456/comments",
+        likes: "/api/posts/123456/likes", 
+        number_of_likes: 0, number_of_comments: 0,
+        state: 'published'
+    };
+    
+    const fakeCallback = function(){
+        return {
+            then: function(callback) {
+                return callback();
+            }
+        };
+    };
+
+    beforeEach(inject(function ($controller, $httpBackend, $mdDialog,
+            AuthService, $rootScope, CommentService, MessageService) {
+        httpBackend = $httpBackend;
+        mdDialog = $mdDialog;
+        commentService = CommentService;
+        messageService = MessageService;
+
+        AuthService.login(user);
+
+        scope = $rootScope.$new();
+        commentCtrl = $controller('CommentController', {
+            scope: scope
+        });
+
+        commentCtrl.user = AuthService.getCurrentUser();
+        commentCtrl.comment = {...comment, likes: [], replies: {}};
+        commentCtrl.post = {...post};
+        commentCtrl.setReplyId();
+        commentCtrl.setCurrentComment();
+    }));
+
+    afterEach(function() {
+        scope.$destroy();
+        comment = {...comment, likes: [], replies: {}};
+        reply = {...reply, likes: []};
+        nilReply = {};
+        httpBackend.verifyNoOutstandingExpectation();
+        httpBackend.verifyNoOutstandingRequest();
+    });
+
```

Comment:
try to test the `onInit function`.
You can create an `init` function, put everything needed there, and call it at `onInit`. its easier to test.